### PR TITLE
Expose the tx mempool monitoring mini protocol in cardano-api

### PR DESCRIPTION
**Pull request by Jimbo4350**
Resolves: #3698

### CODE REVIEW

File: cardano-api/src/Cardano/Api/IPC.hs [hunk 1]
```diff
@@ -550,11 +624,98 @@ submitTxToNodeLocal connctInfo tx = do
...
+
+data LocalTxMonitoringQuery mode
+  -- | Query if a particular tx exists in the mempool. Note that, the absence
+  -- of a transaction does not imply anything about how the transaction was
+  -- processed: it may have been dropped, or inserted in a block.
+  = LocalTxMonitoringQueryTx (TxIdInMode mode)
+  -- | The mempool is modeled as an ordered list of transactions and thus, can
+  -- be traversed linearly. 'LocalTxMonitoringSendNextTx' requests the next transaction from the
+  -- current list. This must be a transaction that was not previously sent to
+  -- the client for this particular snapshot.
```
Comment (newhoggy) [inline cardano-api/src/Cardano/Api/IPC.hs:650]: Is there state associated connection that tracks where the current transaction in the list is?

File: cardano-api/src/Cardano/Api/InMode.hs [hunk 2]
```diff
@@ -6,12 +6,17 @@
 -- | Transactions in the context of a consensus mode, and other types used in
 -- the transaction submission protocol.
 --
-module Cardano.Api.TxInMode (
+module Cardano.Api.InMode (
```
Comment (newhoggy) [inline cardano-api/src/Cardano/Api/InMode.hs:9]: What's the motivation for the module rename?

Maybe the comment for the module needs adjustment too?

File: cardano-api/src/Cardano/Api/IPC.hs (hunk 1 above)
Comment (Jimbo4350) [reply cardano-api/src/Cardano/Api/IPC.hs:650]: There are only 3 available things you can get:
```
data ServerStBusy (kind :: StBusyKind) txid tx slot m a where
  SendMsgReplyNextTx
    :: Maybe tx
    -> ServerStAcquired txid tx slot m a
    -> ServerStBusy NextTx txid tx slot m a

  SendMsgReplyHasTx
    :: Bool
    -> ServerStAcquired txid tx slot m a
    -> ServerStBusy HasTx txid tx slot m a

  SendMsgReplyGetSizes
    :: MempoolSizeAndCapacity
    -> ServerStAcquired txid tx slot m a
    -> ServerStBusy GetSizes txid tx slot m 
    ```

**Review by newhoggy — APPROVED** [review]

**Pull request merged**